guides/BACKEND.md

```diff
@@ -0,0 +1,226 @@
+```
+
+After which you can run:
+
+```bash
+curl -X POST http://localhost:4000/actions/migrateSourceCredAccounts
+```
+
+Which populates it with testing data.
+
+### Configuring Hasura
+
+Before you can create a new table. You'll need to initialize hasura. The same commands are found in the tooling section of the `README.md`.
+
+```bash
+# Start the console
+yarn hasura:console
+
+# Confirm that hasura is running
+yarn hasura migrate status
+...
+```
+
+### Adding a table
+
+- First, navigate to `http://localhost:9695/`. The Hasura Dashboard should appear.
+
+- You can click on the `DATA` tab or navigate to `http://localhost:9695/console/data/schema/public`.
+
+- Click "Create Table"
+
+When creating a table, keep in mind a few things.
+
+1. You should have an `id` for all table relationships and all datatypes should use snake_case.
+
+    - **Example 1:** Table Name: "hello" && Table ID: "hello_id" (as a UUID)
+
+    - **Example 2:** Table Name: "map" && Table ID: "map_id" (as a UUID)
+
+2. You should be mapping objects as foreign keys to their respective UUID. For example if you made a new table that holds a player's messages, it would be something like:
+
+    - From: player_id
+    
+    - To: Player.id
+
+### Updating the GraphQL schemas
+
+Assuming the Docker Containers are running. After creating the table. You will need to run:
+
+```bash
+yarn codegen update-schema
+
+# and then
+yarn codegen generate
+```
+
+This will update the `.graphql` schemas for Hasura and make it easier to create graphql queries for the frontend.
+
+#### Permissions
+
+By default, only admins are allowed to change the permissions. In order to query, add and update data as other than the admin, you'll need to change the permissions.
+
+1. You will need to select the new table you created on the data the Data tab in the Hasura Console.
+
+2. You will need to select the table you created under `Tables/Views/Functions`.
+
+3. Select the permissions tab and then select either `insert`, `select`, `update` and `delete` for `player` or `public`.
+
+4. Generally, you will want to have a custom check. For the most part, data should only be inserted and deleted based on their user id. The following is a custom check you could create for `Row insert permissions` or `Pre-update check` or `Row delete permissions`.
+
+```json
+{
+    "id": {
+        "_eq":"X-Hasura-User-Id"
+    }
+}
+```
+
+5. Furthermore, when selecting or updating data. You can add permissions for specific columns. You select which ones should be allowed via the provided checkboxes.
+
+6. Finally, make sure that the changes for permissions are updated in `hasura/metadata/tables.yaml`.
+`Pre-update check`nsert and read data immediately. There are pre-generated functions that come with Hasura. For creating new entries. The following are example queries you could send immediately to `http://localhost:8080/v1/graphql`.
+
+
+1. Insert a new item
```

Comment:
Should add a title for this section, right now it just goes from `6.` to `1.` . Should also link to Hasura's docs on making queries and mutations for people who want to dive deeper: 
https://hasura.io/docs/1.0/graphql/core/queries/index.html#queries
https://hasura.io/docs/1.0/graphql/core/mutations/index.html